elements/lisk-transactions/src/15_proof_of_misbehavior_transaction.ts

```diff
@@ -0,0 +1,442 @@
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { getAddressFromPublicKey } from '@liskhq/lisk-cryptography';
+import { validator } from '@liskhq/lisk-validator';
+
+import {
+	BaseTransaction,
+	StateStore,
+	StateStorePrepare,
+} from './base_transaction';
+import { convertToAssetError, TransactionError } from './errors';
+import { BlockHeader, TransactionJSON } from './transaction_types';
+import {
+	getBlockBytes,
+	getBlockBytesWithSignature,
+	isPunished,
+	validateSignature,
+} from './utils';
+
+const proofOfMisbehaviorAssetFormatSchema = {
+	type: 'object',
+	required: ['header1', 'header2'],
+	properties: {
+		header1: {
+			type: 'object',
+			required: [
+				'version',
+				'totalAmount',
+				'seedReveal',
+				'totalFee',
+				'reward',
+				'payloadHash',
+				'timestamp',
+				'numberOfTransactions',
+				'payloadLength',
+				'generatorPublicKey',
+				'blockSignature',
+			],
+			properties: {
+				version: {
+					type: 'integer',
+					minimum: 0,
+				},
+				totalAmount: {
+					type: 'string',
+					format: 'amount',
+				},
+				totalFee: {
+					type: 'string',
+					format: 'amount',
+				},
+				reward: {
+					type: 'string',
+					format: 'amount',
+				},
+				seedReveal: {
+					type: 'string',
+					format: 'hex',
+				},
+				payloadHash: {
+					type: 'string',
+					format: 'hex',
+				},
+				timestamp: {
+					type: 'integer',
+					minimum: 0,
+				},
+				numberOfTransactions: {
+					type: 'integer',
+					minimum: 0,
+				},
+				payloadLength: {
+					type: 'integer',
+					minimum: 0,
+				},
+				previousBlockId: {
+					type: ['null', 'string'],
+					format: 'id',
+					minLength: 1,
+					maxLength: 20,
+				},
+				generatorPublicKey: {
+					type: 'string',
+					format: 'publicKey',
+				},
+				maxHeightPrevoted: {
+					type: 'integer',
+					minimum: 0,
+				},
+				maxHeightPreviouslyForged: {
+					type: 'integer',
+					minimum: 0,
+				},
+				height: {
+					type: 'integer',
+					minimum: 1,
+				},
+				blockSignature: {
+					type: 'string',
+					format: 'signature',
+				},
+			},
+		},
+	},
+};
+export interface ProofOfMisbehaviorAsset {
+	readonly header1: BlockHeader;
+	readonly header2: BlockHeader;
+	// tslint:disable-next-line readonly-keyword
+	reward?: bigint;
+}
+
+export class ProofOfMisbehaviorTransaction extends BaseTransaction {
+	public readonly asset: ProofOfMisbehaviorAsset;
+	public static TYPE = 15;
+
+	public constructor(rawTransaction: unknown) {
+		super(rawTransaction);
+		const tx = (typeof rawTransaction === 'object' && rawTransaction !== null
+			? rawTransaction
+			: {}) as Partial<TransactionJSON>;
+		this.asset = (tx.asset || {}) as ProofOfMisbehaviorAsset;
+		this.asset.reward = BigInt(0);
```

Comment:
it should check if `rawTransaction` have it or not, and initialize with it if it has it, otherwise it will be always 0 when you receive from database